daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2083,12 +2083,25 @@ func (devices *DeviceSet) issueDiscard(info *devInfo) error {
 
 // Should be called with devices.Lock() held.
 func (devices *DeviceSet) deleteDevice(info *devInfo, syncDelete bool) error {
+	var deferredRemove bool
+
 	if devices.doBlkDiscard {
 		devices.issueDiscard(info)
 	}
 
 	// Try to deactivate device in case it is active.
-	if err := devices.deactivateDevice(info); err != nil {
+	// If deferred removal is enabled and deferred deletion is disabled
+	// then make sure device is removed synchronously. There have been
+	// some cases of device being busy for short duration and we would
+	// rather busy wait for device removal to take care of these cases.
+	if devices.deferredRemove {
```

Comment:
Makes sense. Will change it. Deferred deletion can't be enabled without deferred removal being on.